catalog/dags/elasticsearch_cluster/healthcheck_dag.py

```diff
@@ -0,0 +1,158 @@
+"""
+Monitor staging and production Elasticsearch cluster health endpoint.
+
+Requests the cluster health and alerts under the following conditions:
+
+- Red cluster health
+- Unexpected number of nodes
+- Unresponsive cluster
+
+Additionally, the DAG will notify (rather than alert) when the cluster health is yellow.
+Yellow cluster health may or may not be an issue, depending on whether it is expected,
+and occurs whenever shards and replicas are being relocated (e.g., during reindexes).
+It is worthwhile to notify in these cases, as an assurance, but we could choose to add
+logic that ignores yellow cluster health during data refresh or other similar operations.
```

Comment:
Given that this DAG is set up to run every 15 minutes (and thus alert every 15 minutes), and that replication of shards for the image index could take several hours, I worry that alerting every 15 minutes for a known state where no action can be taken will generate alert fatigue. The simplest way to combat this might be as you suggest - query if the data refresh is running and skip alerting if the cluster is yellow. Would you mind adding that logic in here?

Reply:
Good point, might as well do it now.

Reply:
Done in the latest commits.